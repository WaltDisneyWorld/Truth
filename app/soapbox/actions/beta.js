import { staticClient } from '../api';

export const FETCH_BETA_PAGE_REQUEST = 'FETCH_BETA_PAGE_REQUEST';
export const FETCH_BETA_PAGE_SUCCESS = 'FETCH_BETA_PAGE_SUCCESS';
export const FETCH_BETA_PAGE_FAIL    = 'FETCH_BETA_PAGE_FAIL';

export function fetchBetaPage(slug = 'index', locale) {
  return (dispatch, getState) => {
    dispatch({ type: FETCH_BETA_PAGE_REQUEST, slug, locale });
    const filename = `${slug}${locale ? `.${locale}` : ''}.html`;
    return staticClient.get(`/instance/beta/${filename}`).then(({ data: html }) => {
      dispatch({ type: FETCH_BETA_PAGE_SUCCESS, slug, locale, html });
      return html;
    }).catch(error => {
      dispatch({ type: FETCH_BETA_PAGE_FAIL, slug, locale, error });
      throw error;
    });
  };
}
