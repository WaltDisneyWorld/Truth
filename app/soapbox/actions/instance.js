import { get } from 'lodash';

import KVStore from 'soapbox/storage/kv_store';
import { getAuthUserUrl } from 'soapbox/utils/auth';
import { parseVersion } from 'soapbox/utils/features';

import api from '../api';

export const INSTANCE_FETCH_REQUEST = 'INSTANCE_FETCH_REQUEST';
export const INSTANCE_FETCH_SUCCESS = 'INSTANCE_FETCH_SUCCESS';
export const INSTANCE_FETCH_FAIL    = 'INSTANCE_FETCH_FAIL';

export const INSTANCE_REMEMBER_REQUEST = 'INSTANCE_REMEMBER_REQUEST';
export const INSTANCE_REMEMBER_SUCCESS = 'INSTANCE_REMEMBER_SUCCESS';
export const INSTANCE_REMEMBER_FAIL    = 'INSTANCE_REMEMBER_FAIL';

export const NODEINFO_FETCH_REQUEST = 'NODEINFO_FETCH_REQUEST';
export const NODEINFO_FETCH_SUCCESS = 'NODEINFO_FETCH_SUCCESS';
export const NODEINFO_FETCH_FAIL    = 'NODEINFO_FETCH_FAIL';

const getMeUrl = state => {
  const me = state.get('me');
  return state.getIn(['accounts', me, 'url']);
};

// Figure out the appropriate instance to fetch depending on the state
export const getHost = state => {
  const accountUrl = getMeUrl(state) || getAuthUserUrl(state);

  try {
    return new URL(accountUrl).host;
  } catch {
    return null;
  }
};

export function rememberInstance(host) {
  return (dispatch, getState) => {
    dispatch({ type: INSTANCE_REMEMBER_REQUEST, host });
    return KVStore.getItemOrError(`instance:${host}`).then(instance => {
      dispatch({ type: INSTANCE_REMEMBER_SUCCESS, host, instance });
      return instance;
    }).catch(error => {
      dispatch({ type: INSTANCE_REMEMBER_FAIL, host, error, skipAlert: true });
    });
  };
}

// We may need to fetch nodeinfo on Pleroma < 2.1
const needsNodeinfo = instance => {
  const v = parseVersion(get(instance, 'version'));
  return v.software === 'Pleroma' && !get(instance, ['pleroma', 'metadata']);
};

export function fetchInstance() {
  return (dispatch, getState) => {
    dispatch({ type: INSTANCE_FETCH_REQUEST });
    return api(getState).get('/api/v1/instance').then(({ data: instance }) => {
      dispatch({ type: INSTANCE_FETCH_SUCCESS, instance });
      if (needsNodeinfo(instance)) {
        dispatch(fetchNodeinfo()); // Pleroma < 2.1 backwards compatibility
      }
    }).catch(error => {
      dispatch({ type: INSTANCE_FETCH_FAIL, error, skipAlert: true });
    });
  };
}

// Tries to remember the instance from browser storage before fetching it
export function loadInstance() {
  return (dispatch, getState) => {
    const host = getHost(getState());

    return dispatch(rememberInstance(host)).finally(() => {
      return dispatch(fetchInstance());
    });
  };
}

export function fetchNodeinfo() {
  return (dispatch, getState) => {
    dispatch({ type: NODEINFO_FETCH_REQUEST });
    api(getState).get('/nodeinfo/2.1.json').then(({ data: nodeinfo }) => {
      dispatch({ type: NODEINFO_FETCH_SUCCESS, nodeinfo });
    }).catch(error => {
      dispatch({ type: NODEINFO_FETCH_FAIL, error, skipAlert: true });
    });
  };
}
