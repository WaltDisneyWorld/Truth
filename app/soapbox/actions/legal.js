import { staticClient } from '../api';

export const FETCH_LEGLA_PAGE_REQUEST = 'FETCH_LEGLA_PAGE_REQUEST';
export const FETCH_LEGLA_PAGE_SUCCESS = 'FETCH_LEGLA_PAGE_SUCCESS';
export const FETCH_LEGLA_PAGE_FAIL    = 'FETCH_LEGLA_PAGE_FAIL';

export function fetchLegalPage(slug = 'index', locale) {
  return (dispatch, getState) => {
    dispatch({ type: FETCH_LEGLA_PAGE_REQUEST, slug, locale });
    const filename = `${slug}${locale ? `.${locale}` : ''}.html`;
    return staticClient.get(`/instance/legal/${filename}`).then(({ data: html }) => {
      dispatch({ type: FETCH_LEGLA_PAGE_SUCCESS, slug, locale, html });
      return html;
    }).catch(error => {
      dispatch({ type: FETCH_LEGLA_PAGE_FAIL, slug, locale, error });
      throw error;
    });
  };
}
