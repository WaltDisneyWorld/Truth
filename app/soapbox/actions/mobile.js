import { staticClient } from '../api';

export const FETCH_MOBILE_PAGE_REQUEST = 'FETCH_MOBILE_PAGE_REQUEST';
export const FETCH_MOBILE_PAGE_SUCCESS = 'FETCH_MOBILE_PAGE_SUCCESS';
export const FETCH_MOBILE_PAGE_FAIL    = 'FETCH_MOBILE_PAGE_FAIL';

export function fetchMobilePage(slug = 'index', locale) {
  return (dispatch, getState) => {
    dispatch({ type: FETCH_MOBILE_PAGE_REQUEST, slug, locale });
    const filename = `${slug}${locale ? `.${locale}` : ''}.html`;
    return staticClient.get(`/instance/mobile/${filename}`).then(({ data: html }) => {
      dispatch({ type: FETCH_MOBILE_PAGE_SUCCESS, slug, locale, html });
      return html;
    }).catch(error => {
      dispatch({ type: FETCH_MOBILE_PAGE_FAIL, slug, locale, error });
      throw error;
    });
  };
}
