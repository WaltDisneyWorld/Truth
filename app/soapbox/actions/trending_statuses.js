import api from '../api';

import { importFetchedStatuses } from './importer';

export const TRENDING_STATUSES_FETCH_REQUEST = 'TRENDING_STATUSES_FETCH_REQUEST';
export const TRENDING_STATUSES_FETCH_SUCCESS = 'TRENDING_STATUSES_FETCH_SUCCESS';
export const TRENDING_STATUSES_FETCH_FAIL    = 'TRENDING_STATUSES_FETCH_FAIL';

export function fetchTrendingStatuses() {
  return (dispatch, getState) => {
    dispatch({ type: TRENDING_STATUSES_FETCH_REQUEST });
    return api(getState).get('/api/v1/truth/trending/truths').then(({ data: statuses }) => {
      dispatch(importFetchedStatuses(statuses));
      dispatch({ type: TRENDING_STATUSES_FETCH_SUCCESS, statuses });
      return statuses;
    }).catch(error => {
      dispatch({ type: TRENDING_STATUSES_FETCH_FAIL, error });
    });
  };
}
