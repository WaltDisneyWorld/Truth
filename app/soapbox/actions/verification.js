import api from '../api';

/**
 * LocalStorage 'soapbox:verification'
 *
 * {
 *   token: String,
 *   challenges: {
 *     email: Number (0 = incomplete, 1 = complete),
 *     sms: Number,
 *     age: Number
 *   }
 * }
 */
const LOCAL_STORAGE_VERIFICATION_KEY = 'soapbox:verification';

const PEPE_FETCH_INSTANCE_SUCCESS = 'PEPE_FETCH_INSTANCE_SUCCESS';
const FETCH_CHALLENGES_SUCCESS = 'FETCH_CHALLENGES_SUCCESS';
const FETCH_TOKEN_SUCCESS = 'FETCH_TOKEN_SUCCESS';

const SET_NEXT_CHALLENGE = 'SET_NEXT_CHALLENGE';
const SET_CHALLENGES_COMPLETE = 'SET_CHALLENGES_COMPLETE';
const SET_LOADING = 'SET_LOADING';

const ChallengeTypes = {
  EMAIL: 'email',
  SMS: 'sms',
  AGE: 'age',
};

/**
 * Fetch the state of the user's verification in local storage.
 *
 * @returns {object}
 * {
 *   token: String,
 *   challenges: {
 *     email: Number (0 = incomplete, 1 = complete),
 *     sms: Number,
 *     age: Number
 *   }
 * }
 */
function fetchStoredVerification() {
  try {
    return JSON.parse(localStorage.getItem(LOCAL_STORAGE_VERIFICATION_KEY));
  } catch {
    return null;
  }
}

/**
 * Remove the state of the user's verification from local storage.
 */
function removeStoredVerification() {
  localStorage.removeItem(LOCAL_STORAGE_VERIFICATION_KEY);
}


/**
 * Fetch and return the Registration token for Pepe.
 * @returns {string}
 */
function fetchStoredToken() {
  try {
    const verification = fetchStoredVerification();
    return verification.token;
  } catch {
    return null;
  }
}

/**
 * Fetch and return the state of the verification challenges.
 * @returns {object}
 * {
 *   challenges: {
 *     email: Number (0 = incomplete, 1 = complete),
 *     sms: Number,
 *     age: Number
 *   }
 * }
 */
function fetchStoredChallenges() {
  try {
    const verification = fetchStoredVerification();
    return verification.challenges;
  } catch {
    return null;
  }
}

/**
 * Update the verification object in local storage.
 *
 * @param {*} verification object
 */
function updateStorage({ ...updatedVerification }) {
  const verification = fetchStoredVerification();

  localStorage.setItem(
    LOCAL_STORAGE_VERIFICATION_KEY,
    JSON.stringify({ ...verification, ...updatedVerification }),
  );
}

/**
 * Fetch Pepe challenges and registration token
 * @returns {promise}
 */
function fetchVerificationConfig() {
  return async(dispatch) => {
    await dispatch(fetchPepeInstance());

    dispatch(fetchRegistrationToken());
  };
}

/**
 * Save the challenges in localStorage.
 *
 * - If the API removes a challenge after the client has stored it, remove that
 *    challenge from localStorage.
 * - If the API adds a challenge after the client has stored it, add that
 *    challenge to localStorage.
 * - Don't overwrite a challenge that has already been completed.
 * - Update localStorage to the new set of challenges.
 *
 * @param {array} challenges - ['age', 'sms', 'email']
 */
function saveChallenges(challenges) {
  const currentChallenges = fetchStoredChallenges() || {};

  const challengesToRemove = Object.keys(currentChallenges).filter((currentChallenge) => !challenges.includes(currentChallenge));
  challengesToRemove.forEach((challengeToRemove) => delete currentChallenges[challengeToRemove]);

  for (let i = 0; i < challenges.length; i++) {
    const challengeName = challenges[i];

    if (typeof currentChallenges[challengeName] !== 'number') {
      currentChallenges[challengeName] = 0;
    }
  }

  updateStorage({ challenges: currentChallenges });
}

/**
 * Finish a challenge.
 * @param {string} challenge - "sms" or "email" or "age"
 */
function finishChallenge(challenge) {
  const currentChallenges = fetchStoredChallenges() || {};
  // Set challenge to "complete"
  currentChallenges[challenge] = 1;

  updateStorage({ challenges: currentChallenges });
}

/**
 * Fetch the next challenge
 * @returns {string} challenge - "sms" or "email" or "age"
 */
function fetchNextChallenge() {
  const currentChallenges = fetchStoredChallenges() || {};
  return Object.keys(currentChallenges).find((challenge) => currentChallenges[challenge] === 0);
}

/**
 * Dispatch the next challenge or set to complete if all challenges are completed.
 */
function dispatchNextChallenge(dispatch) {
  const nextChallenge = fetchNextChallenge();

  if (nextChallenge) {
    dispatch({ type: SET_NEXT_CHALLENGE, challenge: nextChallenge });
  } else {
    dispatch({ type: SET_CHALLENGES_COMPLETE });
  }
}

/**
 * Fetch the challenges and age mininum from Pepe
 * @returns {promise}
 */
function fetchPepeInstance() {
  return (dispatch, getState) => {
    dispatch({ type: SET_LOADING });

    return api(getState).get('/api/v1/pepe/instance').then(response => {
      const { challenges, age_minimum: ageMinimum } = response.data;
      saveChallenges(challenges);
      const currentChallenge = fetchNextChallenge();

      dispatch({ type: PEPE_FETCH_INSTANCE_SUCCESS, instance: { isReady: true, ...response.data } });

      dispatch({
        type: FETCH_CHALLENGES_SUCCESS,
        ageMinimum,
        currentChallenge,
        isComplete: !currentChallenge,
      });
    })
      .finally(() => dispatch({ type: SET_LOADING, value: false }));
  };
}

/**
 * Fetch the regristration token from Pepe unless it's already been stored locally
 * @returns {promise}
 */
function fetchRegistrationToken() {
  return (dispatch, getState) => {
    dispatch({ type: SET_LOADING });

    const token = fetchStoredToken();
    if (token) {
      dispatch({
        type: FETCH_TOKEN_SUCCESS,
        value: token,
      });
      return null;
    }


    return api(getState).post('/api/v1/pepe/registrations')
      .then(response => {
        updateStorage({ token: response.data.access_token });

        return dispatch({
          type: FETCH_TOKEN_SUCCESS,
          value: response.data.access_token,
        });
      })
      .finally(() => dispatch({ type: SET_LOADING, value: false }));
  };
}

function checkEmailAvailability(email) {
  return (dispatch, getState) => {
    dispatch({ type: SET_LOADING });

    const token = fetchStoredToken();

    return api(getState).get(`/api/v1/pepe/account/exists?email=${email}`, {
      headers: { Authorization: `Bearer ${token}` },
    }).finally(() => dispatch({ type: SET_LOADING, value: false }));
  };
}

/**
 * Send the user's email to Pepe to request confirmation
 * @param {string} email
 * @returns {promise}
 */
function requestEmailVerification(email) {
  return (dispatch, getState) => {
    dispatch({ type: SET_LOADING });

    const token = fetchStoredToken();

    return api(getState).post('/api/v1/pepe/verify_email/request', { email }, {
      headers: { Authorization: `Bearer ${token}` },
    })
      .finally(() => dispatch({ type: SET_LOADING, value: false }));
  };
}

/**
 * Confirm the user's email with Pepe
 * @param {string} emailToken
 * @returns {promise}
 */
function confirmEmailVerification(emailToken) {
  return (dispatch, getState) => {
    dispatch({ type: SET_LOADING });

    const token = fetchStoredToken();

    return api(getState).post('/api/v1/pepe/verify_email/confirm', { token: emailToken }, {
      headers: { Authorization: `Bearer ${token}` },
    })
      .then(() => {
        finishChallenge(ChallengeTypes.EMAIL);
        dispatchNextChallenge(dispatch);
      })
      .finally(() => dispatch({ type: SET_LOADING, value: false }));
  };
}

/**
 * Send the user's phone number to Pepe to request confirmation
 * @param {string} phone
 * @returns {promise}
 */
function requestPhoneVerification(phone) {
  return (dispatch, getState) => {
    dispatch({ type: SET_LOADING });

    const token = fetchStoredToken();

    return api(getState).post('/api/v1/pepe/verify_sms/request', { phone }, {
      headers: { Authorization: `Bearer ${token}` },
    })
      .finally(() => dispatch({ type: SET_LOADING, value: false }));
  };
}

/**
 * Confirm the user's phone number with Pepe
 * @param {string} code
 * @returns {promise}
 */
function confirmPhoneVerification(code) {
  return (dispatch, getState) => {
    dispatch({ type: SET_LOADING });

    const token = fetchStoredToken();

    return api(getState).post('/api/v1/pepe/verify_sms/confirm', { code }, {
      headers: { Authorization: `Bearer ${token}` },
    })
      .then(() => {
        finishChallenge(ChallengeTypes.SMS);
        dispatchNextChallenge(dispatch);
      })
      .finally(() => dispatch({ type: SET_LOADING, value: false }));
  };
}

/**
 * Confirm the user's age with Pepe
 * @param {date} birthday
 * @returns {promise}
 */
function verifyAge(birthday) {
  return (dispatch, getState) => {
    dispatch({ type: SET_LOADING });

    const token = fetchStoredToken();

    return api(getState).post('/api/v1/pepe/verify_age/confirm', { birthday }, {
      headers: { Authorization: `Bearer ${token}` },
    })
      .then(() => {
        finishChallenge(ChallengeTypes.AGE);
        dispatchNextChallenge(dispatch);
      })
      .finally(() => dispatch({ type: SET_LOADING, value: false }));
  };
}

/**
 * Create the user's account with Pepe
 * @param {string} username
 * @param {string} password
 * @returns {promise}
 */
function createAccount(username, password) {
  return (dispatch, getState) => {
    dispatch({ type: SET_LOADING });

    const token = fetchStoredToken();

    return api(getState).post('/api/v1/pepe/accounts', { username, password }, {
      headers: { Authorization: `Bearer ${token}` },
    }).finally(() => dispatch({ type: SET_LOADING, value: false }));
  };
}

export {
  PEPE_FETCH_INSTANCE_SUCCESS,
  FETCH_CHALLENGES_SUCCESS,
  FETCH_TOKEN_SUCCESS,
  LOCAL_STORAGE_VERIFICATION_KEY,
  SET_CHALLENGES_COMPLETE,
  SET_LOADING,
  SET_NEXT_CHALLENGE,
  checkEmailAvailability,
  confirmEmailVerification,
  confirmPhoneVerification,
  createAccount,
  fetchStoredChallenges,
  fetchVerificationConfig,
  fetchRegistrationToken,
  removeStoredVerification,
  requestEmailVerification,
  requestPhoneVerification,
  verifyAge,
};
