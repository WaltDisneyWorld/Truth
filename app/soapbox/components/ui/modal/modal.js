import classNames from 'classnames';
import PropTypes from 'prop-types';
import * as React from 'react';
import { defineMessages, useIntl } from 'react-intl';

import Button from '../../button';
import IconButton from '../icon-button/icon-button';

const messages = defineMessages({
  close: { id: 'lightbox.close', defaultMessage: 'Close' },
  confirm: { id: 'confirmations.delete.confirm', defaultMessage: 'Delete' },
});

const Modal = ({
  cancelAction,
  cancelText,
  children,
  confirmationAction,
  confirmationDisabled,
  confirmationText,
  confirmationTheme,
  onClose,
  secondaryAction,
  secondaryText,
  title,
  isScrollable,
}) => {
  const intl = useIntl();
  const buttonRef = React.useRef(null);

  React.useEffect(() => {
    if (buttonRef?.current) {
      buttonRef.current.focus();
    }
  }, [buttonRef]);

  return (
    <div className='inline-block w-full max-w-xl p-6 overflow-hidden text-left align-middle transition-all transform bg-white shadow-xl rounded-2xl pointer-events-auto'>
      <div className='sm:flex sm:items-start w-full justify-between'>
        <div className='w-full'>
          <div className='w-full flex flex-row justify-between items-center'>
            <h3 className='text-lg leading-6 font-medium text-gray-900'>
              {title}
            </h3>

            {onClose && (
              <IconButton
                src={require('@tabler/icons/icons/x.svg')}
                title={intl.formatMessage(messages.close)}
                onClick={onClose}
                className='text-gray-500 hover:text-gray-700'
              />
            )}
          </div>

          <div className={classNames('mt-2 w-full', { 'max-h-96 overflow-y-scroll': isScrollable })}>
            {children}
          </div>
        </div>
      </div>

      {confirmationAction && (
        <div className='mt-5 flex flex-row justify-between'>
          <div className='flex-grow'>
            {cancelAction && (
              <Button
                theme='ghost'
                onClick={cancelAction}
              >
                {cancelText}
              </Button>
            )}
          </div>


          <div className='flex flex-row space-x-2'>
            {secondaryAction && (
              <Button
                theme='secondary'
                onClick={secondaryAction}
              >
                {secondaryText}
              </Button>
            )}

            <Button
              theme={confirmationTheme || 'primary'}
              onClick={confirmationAction}
              disabled={confirmationDisabled}
              ref={buttonRef}
            >
              {confirmationText}
            </Button>
          </div>
        </div>
      )}
    </div>
  );
};

Modal.propTypes = {
  cancelAction: PropTypes.func,
  cancelText: PropTypes.string,
  children: PropTypes.node,
  confirmationAction: PropTypes.func,
  confirmationDisabled: PropTypes.bool,
  confirmationText: PropTypes.string,
  confirmationTheme: PropTypes.oneOf(['danger']),
  onClose: PropTypes.func.isRequired,
  secondaryAction: PropTypes.func,
  secondaryText: PropTypes.string,
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  isScrollable: PropTypes.bool,
};

export default Modal;
