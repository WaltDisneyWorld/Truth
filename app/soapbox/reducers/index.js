import { Map as ImmutableMap } from 'immutable';
import { combineReducers } from 'redux-immutable';

import { AUTH_LOGGED_OUT } from 'soapbox/actions/auth';
import { NODE_ENV } from 'soapbox/build_config';

import accounts from './accounts';
import accounts_counters from './accounts_counters';
import accounts_meta from './accounts_meta';
import admin from './admin';
import admin_log from './admin_log';
import alerts from './alerts';
import aliases from './aliases';
import auth from './auth';
import backups from './backups';
import chat_message_lists from './chat_message_lists';
import chat_messages from './chat_messages';
import chats from './chats';
import compose from './compose';
import contexts from './contexts';
import conversations from './conversations';
import custom_emojis from './custom_emojis';
import domain_lists from './domain_lists';
import dropdown_menu from './dropdown_menu';
import filters from './filters';
import group_editor from './group_editor';
import group_lists from './group_lists';
import group_relationships from './group_relationships';
import groups from './groups';
import height_cache from './height_cache';
import identity_proofs from './identity_proofs';
import instance from './instance';
import listAdder from './list_adder';
import listEditor from './list_editor';
import lists from './lists';
import me from './me';
import meta from './meta';
import modal from './modal';
import mutes from './mutes';
import notifications from './notifications';
import patron from './patron';
import pending_statuses from './pending_statuses';
import polls from './polls';
import profile_hover_card from './profile_hover_card';
import push_notifications from './push_notifications';
import relationships from './relationships';
import reports from './reports';
import scheduled_statuses from './scheduled_statuses';
import search from './search';
import security from './security';
import settings from './settings';
import sidebar from './sidebar';
import soapbox from './soapbox';
import status_lists from './status_lists';
import statuses from './statuses';
import suggestions from './suggestions';
import timelines from './timelines';
import trending_statuses from './trending_statuses';
import trends from './trends';
import user_lists from './user_lists';
import verification from './verification';

const appReducer = combineReducers({
  dropdown_menu,
  timelines,
  meta,
  alerts,
  modal,
  user_lists,
  domain_lists,
  status_lists,
  accounts,
  accounts_counters,
  statuses,
  relationships,
  settings,
  push_notifications,
  mutes,
  reports,
  contexts,
  compose,
  search,
  notifications,
  height_cache,
  custom_emojis,
  identity_proofs,
  lists,
  listEditor,
  listAdder,
  filters,
  conversations,
  suggestions,
  polls,
  trends,
  groups,
  group_relationships,
  group_lists,
  group_editor,
  sidebar,
  patron,
  soapbox,
  instance,
  me,
  auth,
  admin,
  chats,
  chat_messages,
  chat_message_lists,
  profile_hover_card,
  backups,
  admin_log,
  security,
  scheduled_statuses,
  pending_statuses,
  aliases,
  accounts_meta,
  trending_statuses,
  verification,
});

// Clear the state (mostly) when the user logs out
const logOut = (state = ImmutableMap()) => {
  if (NODE_ENV === 'production') {
    location.href = '/login';
  }

  const whitelist = ['instance', 'soapbox', 'custom_emojis', 'auth'];

  return ImmutableMap(
    whitelist.reduce((acc, curr) => {
      acc[curr] = state.get(curr);
      return acc;
    }, {}),
  );
};

const rootReducer = (state, action) => {
  switch(action.type) {
  case AUTH_LOGGED_OUT:
    return appReducer(logOut(state), action);
  default:
    return appReducer(state, action);
  }
};

export default rootReducer;
