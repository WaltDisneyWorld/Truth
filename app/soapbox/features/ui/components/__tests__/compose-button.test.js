import { Map as ImmutableMap } from 'immutable';
import React from 'react';

import { MODAL_OPEN } from 'soapbox/actions/modal';
import rootReducer from 'soapbox/reducers';
import { createComponent, mockStore } from 'soapbox/test_helpers';


import ComposeButton from '../compose_button';

describe('<ComposeButton />', () => {
  it('renders a button element', () => {
    const component = createComponent(<ComposeButton />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('dispatches the MODAL_OPEN action', () => {
    const store = mockStore(rootReducer(ImmutableMap(), {}));
    const component = createComponent(<ComposeButton />, { store });

    expect(store.getActions().length).toEqual(0);
    component.root.findByType('button').props.onClick();
    expect(store.getActions()[0].type).toEqual(MODAL_OPEN);
  });
});
