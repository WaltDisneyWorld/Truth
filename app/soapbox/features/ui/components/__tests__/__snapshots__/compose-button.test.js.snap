// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComposeButton /> renders a button element 1`] = `
<div
  className="mt-4"
>
  <button
    className="inline-flex items-center border font-medium rounded-full focus:outline-none appearance-none transition-all border-transparent text-white bg-accent-500 hover:bg-accent-300 focus:ring-pink-500 focus:ring-2 focus:ring-offset-2 px-6 py-3 text-base flex w-full justify-center"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <div
      className="svg-icon mr-2"
    >
      <svg
        id={
          Object {
            "process": [Function],
          }
        }
      />
    </div>
    <span>
      Compose
    </span>
  </button>
</div>
`;
